### PR TITLE
Spotify Token Fix

### PR DESCRIPTION
**Pull request by Byteingpython**
## Checklist

 - [x] Read [Contribution Guidelines](https://github.com/SlimeCloud/java-SlimeBot#-contributing) and [Style Guides](https://github.com/SlimeCloud/java-SlimeBot#-style-guide)
 - [x] Test the Code
 - [x] This PR is ready to review and merge

## Description
The spotify Api instance is now created new every time the episodes are fetched to avoid token expiration

## Related Issue
#112

### CODE REVIEW

**Review by MineKing9534 — CHANGES_REQUESTED** [review]
You should rather only get a new access token without creating an entire new instance of the spotify api class. Simple put the setup code back into the constructor and only call the setAccessToken part multiple times. Also, instead of getting a new token for every request you should rather save the expire of the token and only request a new one if the old one is expired. To do so, you can store a long that holds the expiry timestamp. Every time you get a new token, you set this long to `System.currentTimeMillis() + token.getExpiresIn() * 1000` and only update the token `if(System.currentTimeMillis() > tokenExpiry`.

File: src/main/java/com/slimebot/alerts/spotify/SpotifyListener.java [hunk 1]
```diff
@@ -120,4 +120,14 @@ private void broadcast(String format, Function<SpotifyNotificationConfig, Option
 	public static Logger getLogger() {
 		return logger;
 	}
+
+	private void fetchToken(SpotifyApi api){
```
Comment (MineKing9534) [inline src/main/java/com/slimebot/alerts/spotify/SpotifyListener.java:124]: Why do you have the SpotifyApi as parameter? It is available as field in the class

File: src/main/java/com/slimebot/alerts/spotify/SpotifyListener.java [hunk 2]
```diff
@@ -23,26 +24,25 @@
 
 @Slf4j
 public class SpotifyListener {
+	private long tokenExpiry=0;
```
Comment (MineKing9534) [inline src/main/java/com/slimebot/alerts/spotify/SpotifyListener.java:27]: Formatting should be `tokenExpiry = 0`.

File: src/main/java/com/slimebot/alerts/spotify/SpotifyListener.java [hunk 3]
```diff
@@ -23,26 +24,25 @@
...
-		}
 	}
-
 	public void register() {
 		Main.scheduleAtFixedRate(1, TimeUnit.HOURS, this::check);
 	}
 
+
 	public void check() {
+		if(System.currentTimeMillis()>tokenExpiry){
```
Comment (MineKing9534) [inline src/main/java/com/slimebot/alerts/spotify/SpotifyListener.java:43]: Missing spaces: `if (.. > ..)`
Don't use `{}` for single line blocks

File: src/main/java/com/slimebot/alerts/spotify/SpotifyListener.java [hunk 4]
```diff
@@ -23,26 +24,25 @@
 
 @Slf4j
 public class SpotifyListener {
+	private long tokenExpiry=0;
 
 	private final SpotifyApi api;
 
-	public SpotifyListener() {
-		api = new SpotifyApi.Builder()
+	public SpotifyListener(){
```
Comment (MineKing9534) [inline src/main/java/com/slimebot/alerts/spotify/SpotifyListener.java:31]: Missing space between `()` and `{`

**Review by MineKing9534 — CHANGES_REQUESTED** [review]

**Review by MineKing9534 — APPROVED** [review]

**Review by MatitoDA — APPROVED** [review]

**Pull request merged**